apis/nucleus/handler/hypercube-handler.js

```diff
@@ -41,30 +55,70 @@ class HyperCubeHandler extends DataPropertyHandler {
     // Set auto-sort property (compatibility 0.85 -> 0.9), can probably be removed in 1.0
     this.hcProperties.qDimensions = hcUtils.setFieldProperties(this.hcProperties.qDimensions);
     this.hcProperties.qMeasures = hcUtils.setFieldProperties(this.hcProperties.qMeasures);
-    return {};
   }
 
   // ----------------------------------
   // ----------- DIMENSIONS -----------
   // ----------------------------------
 
+  /**
+   * @returns {NxDimension[]} dimensions
+   * @description Returns the dimensions of the hypercube.
+   * @memberof HyperCubeHandler
+   * @example
+   * const dimensions = hyperCubeHandler.getDimensions();
+   */
   getDimensions() {
     return this.hcProperties ? this.hcProperties.qDimensions : [];
   }
 
+  /**
+   * @returns {NxDimension[]} alternative dimensions
+   * @description Returns the alternative dimensions of the hypercube.
+   * @memberof HyperCubeHandler
+   * @example
+   * const alternativeDimensions = hyperCubeHandler.getAlternativeDimensions();
+   */
   getAlternativeDimensions() {
     return this.hcProperties?.qLayoutExclude?.qHyperCubeDef?.qDimensions ?? [];
   }
 
+  /**
+   * @param {string} cId
+   * @returns {NxDimensionInfo} dimension layout
+   * @description Returns the dimension layout of the hypercube for a given cId.
+   * @memberof HyperCubeHandler
+   * @example
+   * const dimensionLayout = hyperCubeHandler.getDimensionLayout(cId);
+   */
   getDimensionLayout(cId) {
     return this.getDimensionLayouts().filter((item) => cId === item.cId)[0];
   }
 
+  /**
+   * @returns {NxDimensionInfo[]} dimension layouts
+   * @description Returns the dimension layouts of the hypercube.
+   * @memberof HyperCubeHandler
+   * @example
+   * const dimensionLayouts = hyperCubeHandler.getDimensionLayouts();
+   */
   getDimensionLayouts() {
     const hc = hcUtils.getHyperCube(this.layout, this.path);
     return hc ? hc.qDimensionInfo : [];
   }
 
+  /**
+   *
+   * @param {object} dimension
+   * @param {boolean} alternative
+   * @param {number} idx
```

Comment:
mark as optional (?)
```suggestion
   * @param {number} [idx]
```